crypto/crypto.go

```diff
@@ -55,7 +61,23 @@ func (c crypto) CreateQuorumCert(block *hotstuff.Block, signatures []hotstuff.Pa
 	if err != nil {
 		return hotstuff.QuorumCert{}, err
 	}
-	return hotstuff.NewQuorumCert(sig, block.View(), block.Hash()), nil
+	latencyVector := c.prepareLatencyVector(block, signatures)
+	return hotstuff.NewQuorumCert(c.opts.ID(), sig, block.View(), block.Hash(), latencyVector), nil
+}
+
+func (c crypto) prepareLatencyVector(block *hotstuff.Block, signatures []hotstuff.PartialCert) map[uint32]uint64 {
+	latencyVector := make(map[uint32]uint64)
+	if !c.opts.IsLatencyCalculationEnabled() || c.opts.ID() != block.Proposer() {
+		return latencyVector
+	}
+
+	for _, sig := range signatures {
+		//timeDiff := sig.Time().UnixNano() - block.Time().UnixNano()
+		//timeDiff := sig.Time().UnixMilli() - block.Time().UnixMilli()
+		timeDiff := sig.Time().UnixMicro() - block.Time().UnixMicro()
+		latencyVector[uint32(sig.Signer())] = uint64(timeDiff)
```

Comment:
Looks like this is the place to encode the latency into 32 or 24 bits and encode the signer with 32 or 8 bits, depending on the choice of proto representation for the latency vector. See other comment.